pkg/diskmaker/controllers/lv/add.go

```diff
@@ -0,0 +1,161 @@
+		// InformerFactory: , // unused
+
+	}
+
+	r := &ReconcileLocalVolume{
+		client:          mgr.GetClient(),
+		scheme:          mgr.GetScheme(),
+		eventSync:       newEventReporter(mgr.GetEventRecorderFor(ComponentName)),
+		symlinkLocation: common.GetLocalDiskLocationPath(),
+		cleanupTracker:  cleanupTracker,
+		runtimeConfig:   runtimeConfig,
+		deleter:         provDeleter.NewDeleter(runtimeConfig, cleanupTracker),
+		provisionerMux:  provisionerMux,
+	}
+	// Create a new controller
+	//	apis.AddToScheme(r.scheme)
+	c, err := controller.New(ComponentName, mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	err = c.Watch(&source.Kind{Type: &corev1.ConfigMap{}}, &handler.EnqueueRequestForOwner{
+		OwnerType: &localv1.LocalVolume{},
+	})
+
+	err = c.Watch(&source.Kind{Type: &localv1.LocalVolume{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	// TODO enqueue for the PV based on labels
+
+	// update owned-pv cache used by provisioner/deleter libs and enequeue owning lvset
+	// only the cache is touched by
+	err = c.Watch(&source.Kind{Type: &corev1.PersistentVolume{}}, &handler.Funcs{
+		GenericFunc: func(e event.GenericEvent, q workqueue.RateLimitingInterface) {
+			pv, ok := e.Object.(*corev1.PersistentVolume)
+			if ok {
+				handlePVChange(runtimeConfig, pv, q)
+			}
+		},
+		CreateFunc: func(e event.CreateEvent, q workqueue.RateLimitingInterface) {
+			pv, ok := e.Object.(*corev1.PersistentVolume)
+			if ok {
+				handlePVChange(runtimeConfig, pv, q)
+			}
+		},
+		UpdateFunc: func(e event.UpdateEvent, q workqueue.RateLimitingInterface) {
+			pv, ok := e.ObjectNew.(*corev1.PersistentVolume)
+			if ok {
+				handlePVChange(runtimeConfig, pv, q)
+			}
+		},
+		DeleteFunc: func(e event.DeleteEvent, q workqueue.RateLimitingInterface) {
+			pv, ok := e.Object.(*corev1.PersistentVolume)
+			if ok {
+				handlePVChange(runtimeConfig, pv, q)
+			}
+		},
+	})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func handlePVChange(runtimeConfig *provCommon.RuntimeConfig, pv *corev1.PersistentVolume, q workqueue.RateLimitingInterface) {
+	// skip non-owned PVs
+	name, found := pv.Annotations[provCommon.AnnProvisionedBy]
+	if !found || name != runtimeConfig.Name {
+		return
+	}
+
+	// enqueue owner
+	ownerName, found := pv.Labels[common.PVOwnerNameLabel]
+	if !found {
+		return
+	}
+	ownerNamespace, found := pv.Labels[common.PVOwnerNameLabel]
+	if !found {
+		return
+	}
+	ownerKind, found := pv.Labels[common.PVOwnerKindLabel]
+	if ownerKind != localv1.LocalVolumeKind || !found {
+		return
+	}
+
+	q.Add(reconcile.Request{NamespacedName: types.NamespacedName{Name: ownerName, Namespace: ownerNamespace}})
+
+}
+
+type ReconcileLocalVolume struct {
+	client          client.Client
+	scheme          *runtime.Scheme
+	symlinkLocation string
+	localVolume     *localv1.LocalVolume
+	eventSync       *eventReporter
+
+	// static-provisioner stuff
+	provisionerMux *sync.Mutex
```

Comment:
by default maximum number of concurrent reconciles is - 1 in controller-runtime. Do we still need this mutex?

Reply:
Removed